internal/provider/providers/aws/provider.go

```diff
@@ -0,0 +1,184 @@
+	// in the internal/provider/errors package, or add your own
+	// if really necessary. When returning an error, always use
+	// fmt.Errorf (to enforce the caller to use errors.Is()).
+	switch {
+	case domain == "":
+		return fmt.Errorf("%w", errors.ErrDomainNotSet)
+	case host == "":
+		return fmt.Errorf("%w", errors.ErrHostNotSet)
+	case providerSpecificSettings.AwsAccessKey == "":
+		return fmt.Errorf("%w", errors.ErrAccessKeyIDNotSet)
+	case providerSpecificSettings.AwsSecretAccessKey == "":
+		return fmt.Errorf("%w", errors.ErrSecretAccessKeyNotSet)
+	case providerSpecificSettings.HostedZoneID == "":
+		return fmt.Errorf("%w", errors.ErrHostedZoneIDNotSet)
+	}
+	return nil
+}
+
+func (p *Provider) String() string {
+	return utils.ToString(p.domain, p.host, constants.AWS, p.ipVersion)
+}
+
+func (p *Provider) Domain() string {
+	return p.domain
+}
+
+func (p *Provider) Host() string {
+	return p.host
+}
+
+func (p *Provider) IPVersion() ipversion.IPVersion {
+	return p.ipVersion
+}
+
+func (p *Provider) IPv6Suffix() netip.Prefix {
+	return p.ipv6Suffix
+}
+
+func (p *Provider) Proxied() bool {
+	return false
+}
+
+func (p *Provider) BuildDomainName() string {
+	return utils.BuildDomainName(p.host, p.domain)
+}
+
+func (p *Provider) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain:    fmt.Sprintf("<a href=\"http://%s\">%s</a>", p.BuildDomainName(), p.BuildDomainName()),
+		Host:      p.Host(),
+		Provider:  "<a href=\"https://aws.amazon.com/route53/\">Amazon Route 53</a>",
+		IPVersion: p.ipVersion.String(),
+	}
+}
+
+func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Addr) (newIP netip.Addr, err error) {
+	// API details https://docs.aws.amazon.com/Route53/latest/APIReference/API_ChangeResourceRecordSets.html
+	// GO API https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/route53#Client.ChangeTagsForResource
+
+	recordType := types.RRTypeA
+	if p.ipVersion == ipversion.IP6 {
+		recordType = types.RRTypeAaaa
+	}
+
+	input := &route53.ChangeResourceRecordSetsInput{
+		ChangeBatch: &types.ChangeBatch{
+			Changes: []types.Change{
+				{
+					Action: types.ChangeActionUpsert, // Uses update or insert feature from Route53
+					ResourceRecordSet: &types.ResourceRecordSet{
+						Name: aws.String(p.BuildDomainName()),
+						Type: recordType,
+						ResourceRecords: []types.ResourceRecord{
+							{
+								Value: aws.String(ip.String()),
+							},
+						},
+						TTL: aws.Int64(p.ttl),
+						// SetIdentifier not set means Simple Routing,
+					},
+				},
+			},
+			Comment: aws.String("Record updated by ddns-updater"),
+		},
+		HostedZoneId: aws.String(p.hostedZoneID),
+	}
+
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(p.awsAccessKeyID, p.awsSecretAccessKey, "")),
+		config.WithRegion(p.region),
+	)
+	if err != nil {
+		return netip.Addr{}, fmt.Errorf("failed loading default config %w", err)
```

Comment:
nit don't prefix error wrappings with `failed to`  or `error ...`, since with multiple levels it repeats itself i.e. `failed to updated: failed to do this: failed to do that: actual error`

```suggestion
		return netip.Addr{}, fmt.Errorf("loading default config %w", err)
```